src/olympia/scanners/tests/test_tasks.py

```diff
@@ -612,6 +613,72 @@ def test_does_not_run_when_results_contain_errors(self, yara_compile_mock):
         # The task should always return the results.
         assert received_results == self.results
 
+    @mock.patch('olympia.scanners.tasks.statsd.incr')
+    @override_switch('yara-read-binary', active=True)
+    def test_run_in_binary_mode(self, incr_mock):
+        self.upload = self.get_upload('webextension_with_image.zip')
+
+        assert len(ScannerResult.objects.all()) == 0
+        # This rule will match for all files in the xpi.
+        rule = ScannerRule.objects.create(
+            name='match_png',
+            scanner=YARA,
+            definition='rule match_png { '
+                       'strings: $png = { 89 50 4E 47 0D 0A 1A 0A } '
+                       'condition: $png at 0 }',
+        )
+
+        received_results = run_yara(self.results, self.upload.pk)
+
+        yara_results = ScannerResult.objects.all()
+        assert len(yara_results) == 1
+        yara_result = yara_results[0]
+        assert yara_result.upload == self.upload
+        assert len(yara_result.results) == 1
+        assert yara_result.results[0] == {
+            'rule': rule.name,
+            'tags': [],
+            'meta': {'filename': 'img.png'},
+        }
+        assert incr_mock.called
```

Comment:
We don't necessarily need to mock/assert statsd stuff, let's keep this test case focused on one thing: reading binary content correctly when the switch is enabled.

Reply:
e2bfd9c225